src/units.h

```diff
@@ -721,4 +721,51 @@ T read_from_json_string( JsonIn &jsin, const std::vector<std::pair<std::string,
     return result;
 }
 
+template<typename T>
+void dump_to_json_string( T t, JsonOut &jsout,
+                          const std::vector<std::pair<std::string, T>> &units )
+{
+    // deduplicate unit strings and choose the shortest representations
+    std::map<T, std::string> sorted_units;
+    for( const auto &p : units ) {
+        const auto it = sorted_units.find( p.second );
+        if( it != sorted_units.end() ) {
+            if( p.first.length() < it->second.length() ) {
+                it->second = p.first;
+            }
+        } else {
+            sorted_units.emplace( p.second, p.first );
+        }
+    }
+    std::string str;
+    bool written = false;
+    for( auto it = sorted_units.rbegin(); it != sorted_units.rend(); ++it ) {
+        const int val = static_cast<int>( t / it->first );
+        if( val != 0 ) {
+            if( written ) {
+                str += ' ';
+            }
+            int tmp = val;
+            if( tmp < 0 ) {
+                str += '-';
+                tmp = -tmp;
+            }
+            const size_t val_beg = str.size();
+            while( tmp != 0 ) {
+                str += static_cast<char>( '0' + tmp % 10 );
+                tmp /= 10;
+            }
+            std::reverse( str.begin() + val_beg, str.end() );
```

Comment:
Can't you use `std::to_string( val )` to convert the int to a decimal number string?

Reply:
The result of to_string depends on the locale, and it could insert thousand separaters, which we do not want.

Reply:
You can use `ostringstream`, because that can be imbued with the default locale.  It might be slower at runtime, but it would be a lot less code.  (That's effectively what the existing `JsonOut` `int` writing code does).